include/tango/server/device_6.h

```diff
@@ -0,0 +1,75 @@
+//===================================================================================================================
+//
+// file :		device_6.h
+//
+// project :		tango
+//
+// author(s) :		ingvord
+//
+// Copyright (C) :      2004-2017
+//						European Synchrotron Radiation Facility
+//                      BP 220, Grenoble 38043
+//                      FRANCE
+//
+// This file is part of Tango.
+//
+// Tango is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Tango is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+// of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License along with Tango.
+// If not, see <http://www.gnu.org/licenses/>.
+//===================================================================================================================
+#ifndef TANGO_DEVICE_6_H
+#define TANGO_DEVICE_6_H
+
+#include <tango.h>
+
+namespace Tango {
+
+    void handle_cmd_info_enum_labels(DevVarStringArray *, const vector<string> &);
+
+    class Device_6Impl : public virtual POA_Tango::Device_6,
+                         public Device_5Impl {
+    public:
+        Device_6Impl(DeviceClass *device_class, string &dev_name) : Device_5Impl(device_class, dev_name) {
+            idl_version = 6;
+        }
+
+        Device_6Impl(DeviceClass *device_class, string &dev_name, string &desc) : Device_5Impl(device_class, dev_name,
+                                                                                               desc) {
+            idl_version = 6;
+        }
+
+        Device_6Impl(DeviceClass *device_class, string &dev_name, string &desc, DevState dev_state, string &dev_status)
+                : Device_5Impl(device_class, dev_name, desc, dev_state, dev_status) {
+            idl_version = 6;
+        }
+
+        Device_6Impl(DeviceClass *device_class, const char *dev_name, const char *desc, DevState dev_state,
```

Comment:
In Device_5Impl version, there are some default values defined for desc, dev_state and dev_status arguments.
This is important if you want to be backward compatible with servers generated with Pogo which generates code like the following for a Tango class named TestCmds:

> TestCmds::TestCmds(Tango::DeviceClass *cl, const char *s, const char *d)
>  : TANGO_BASE_CLASS(cl, s, d)
> {
> 	/*----- PROTECTED REGION ID(TestCmds::constructor_3) ENABLED START -----*/
> 	init_device();
> 	
> 	/*----- PROTECTED REGION END -----*/	//	TestCmds::constructor_3
> }